nixos/modules/services/desktops/geoclue2.nix

```diff
@@ -265,7 +341,23 @@ in
           submission-url = cfg.submissionUrl;
           submission-nick = cfg.submissionNick;
         };
+        compass = {
+          enable = cfg.enableCompass;
+        };
+        static-source = {
+          enable = cfg.enableStaticSource;
+        };
       } // mapAttrs' appConfigToINICompatible cfg.appConfig);
+
+    environment.etc."geolocation" = {
+      enable = cfg.enableStaticSource;
+      user = "geoclue";
+      group = "geoclue";
+      mode = "0600";
```

Comment:
This probably won't help with "extra security" because the contents of the file will still exist, world readable, in `/nix/store`.

Reply:
Yes, this is true. In fact it does nothing, because if the file is created from text, then it's world readable, and if it's a symlink to another file, then the permissions aren't modified. I'll keep it for now, since there's no harm.